File: sdks/typescript/src/limits.ts
export const MAX_WS_MESSAGE_SIZE = 64 * 1024;
export const MAX_RTC_CONTROL_SIZE = 64 * 1024;
export const MAX_RTC_STREAM_SIZE = 16 * 1024;
export const MAX_PRESENCE_SIZE = 8 * 1024;
export const MAX_DATA_VALUE_SIZE = 256 * 1024;
export const MAX_TOPIC_NAME_LENGTH = 128;
export const MAX_CLIENT_META_SIZE = 16 * 1024;

export function validatePayloadSize(json: string, limit: number, label: string): void {
  const size = new TextEncoder().encode(json).byteLength;
  if (size > limit) {
    throw new Error(`${label} exceeds size limit: ${size} bytes > ${limit} bytes`);
  }
}

export function validateTopicName(topic: string): void {
  if (topic.length > MAX_TOPIC_NAME_LENGTH) {
    throw new Error(`Topic name exceeds ${MAX_TOPIC_NAME_LENGTH} characters: "${topic}"`);
  }
}

File: sdks/swift/Sources/StarfishClient/Limits.swift
import Foundation

public enum Limits {
    public static let maxWSMessageSize = 64 * 1024
    public static let maxRTCControlSize = 64 * 1024
    public static let maxRTCStreamSize = 16 * 1024
    public static let maxPresenceSize = 8 * 1024
    public static let maxDataValueSize = 256 * 1024
    public static let maxTopicNameLength = 128
    public static let maxClientMetaSize = 16 * 1024
}

func validatePayloadSize(_ json: String, limit: Int, label: String) throws {
    let size = json.utf8.count
    if size > limit {
        throw StarfishError(
            code: .payloadTooLarge,
            message: "\(label) exceeds size limit: \(size) bytes > \(limit) bytes"
        )
    }
}

func validateTopicName(_ topic: String) throws {
    if topic.count > Limits.maxTopicNameLength {
        throw StarfishError(
            code: .topicNameTooLong,
            message: "Topic name exceeds \(Limits.maxTopicNameLength) characters: \"\(topic)\""
        )
    }
}
